be/src/olap/rowset/segment_v2/column_writer.cpp

```diff
@@ -530,7 +530,12 @@ Status ScalarColumnWriter::append_data(const uint8_t** ptr, size_t num_rows) {
         remaining -= num_written;
 
         if (_page_builder->is_page_full()) {
-            RETURN_IF_ERROR(finish_current_page());
+            // get next data for next array_item_rowid
+            if (remaining == 0) {
+                RETURN_IF_ERROR(finish_current_page());
+            } else {
+                RETURN_IF_ERROR(finish_current_page(*ptr));
```

Comment:
It's very tricky to use *ptr as next offset. It can be used to verify the bug, but not suitable for production code.